ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSerdeFactory.java

```diff
@@ -145,18 +158,12 @@ private <T> KsqlConnectDeserializer<T> createDeserializer(
 
   private DataTranslator getDataTranslator(final Schema schema,
       final Optional<Schema> physicalSchema, final boolean isDeserializer) {
-    // If physical schema exists, we use physical schema to translate to connect data. During
-    // deserialization, if physical schema exists, we use original schema to translate to ksql data.
-    return physicalSchema.<DataTranslator>map(
-            value -> isDeserializer ? new ConnectDataTranslator(schema)
-                : new ConnectSRSchemaDataTranslator(value))
```

Comment:
Why not call `ConnectSRSchemaDataTranslator` for `schemaId` case? We don't want to translate `physcialSchema` obtained from `schemaId` to compatible one and use that to do serialization. We want to use raw `physcialSchema` to do serialization if `schemaId` is provided.

Reply:
The `ConnectSRSchemaDataTranslator` does not work either. I just made a fix and created a `ProtobufSRSchemaDataTranslator` for protobuf only. Once I understand all these formats, I will put everything into one class (in a follow-up PR)